torch/_inductor/compile_fx.py

```diff
@@ -583,126 +675,87 @@ def compile_fx_inner(
         return compiled_graph
 
     if cudagraphs:
-        # output args are tuple of first argument
-        output = output_node(gm)
-        assert len(output.args) == 1
-        stack_traces = [
-            (arg.stack_trace if isinstance(arg, torch.fx.node.Node) else None)
-            for arg in output.args[0]
-        ]
-
-        complex_memory_overlap_inputs = any(
-            complex_memory_overlap(t)
-            for t in example_inputs
-            if isinstance(t, torch.Tensor)
-        )
-
-        if not config.triton.cudagraph_support_input_mutation:
-            # Skip supports for cudagraph-managed tensors
-            from torch._inductor.cudagraph_utils import (
-                check_for_mutation_ignore_cuda_graph_managed_tensor,
-            )
-
-            has_mutation_str = check_for_mutation_ignore_cuda_graph_managed_tensor(
-                gm, compiled_graph, static_input_idxs
-            )
-            has_mutation = has_mutation_str is not None
-
-            if has_mutation:
-                compiled_graph.disabled_cudagraphs_reason = has_mutation_str
+        # check cudagraph disabling reasons from inductor lowering
+        if compiled_graph.disabled_cudagraphs_reason:
+            if "cuda" in compiled_graph.device_types:
+                log_cudagraph_skip_and_bump_counter(
+                    f"skipping cudagraphs due to {compiled_graph.disabled_cudagraphs_reason}"
+                )
+            else:
+                counters["inductor"]["cudagraph_skips"] += 1
+            BoxedBool.disable(cudagraphs)
```

Comment:
I really want to just use a boolean instead of a BoxedBool side effect here, but I don't know the callers of compile_fx_inner well enough to know whether it depends on this boolean being mutated. So for now I'm leaving it. Instead, I think it would be better refactored so that we use a local boolean like "ran_cudagraphs", so we don't need to explicitly understand which inputs are being mutated by these function calls.

Reply:
There's a shared boolean between the forward and backward, and we disable the backward if the forward didnt run (too much copying of activations)

Reply:
Would it be possible to add a test for cache hit forward, cache miss backward where we check we disable cudagraphs ?